exonum-java-binding/doc/Migration_guide_0.9.md

```diff
@@ -0,0 +1,179 @@
+# Migrating Java Services to 0.9
+
+The flagship feature of Exonum Java 0.9 release is support for dynamic services&nbsp;—
+services that can be added to the blockchain network after it has been started;
+and multiple instances of the same service.
+
+This guide provides instructions on updating existing services to be compatible with the new
+and updated features.
+
+## Dynamic Services 101
+
+Dynamic Java services are packaged in an _artifact_: a JAR archive with some metadata. An Exonum
+artifact has a runtime-specific _name_ and globally-unique _identifier_. Java services must
```

Comment:
The artifact id story is confusing because there are several conflicting terms:
* '_Exonum_ artifact id' (runtime_id + name, e.g., runtime_id=0, name=com.exonum.example:timestamping:1.0.2) vs '_Maven_ artifact id' (timestamping).
* 'Exonum artifact name' (e.g. com.exonum.example:timestamping:1.0.2 or rust-timestamping:3.2.1 or anything, actually) vs 'Maven artifact id' (which could be considered as project _name_).

I am not sure we shall explain 'identifier' here at all if it is not used during deploy/instantiation directly. (@vitvakatu is it?)